orca-bakery/src/test/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStageSpec.groovy

```diff
@@ -154,6 +154,51 @@ class BakeStageSpec extends Specification {
     thrown(ConstraintViolationException)
   }
 
+  def "should NOT fail if image names from unrelated bake stages don't match"() {
+    given:
+    def pipeline = pipeline {
+      stage {
+        id = "1"
+        type = "bake"
+        context = [
+          "region": "us-east-1",
+          "regions": ["us-east-1", "us-west-2"]
+        ]
+        status = ExecutionStatus.RUNNING
+      }
+      // this is a sibling bake stage whose child bake contexts should not be included in stage 1's outputs, but are
+      stage {
+        id = "2"
+        type = "bake"
+        context = [
+          "region": "us-east-1",
+          "regions": ["eu-east-1"]
+        ]
+        status = ExecutionStatus.RUNNING
+      }
+    }
+
+    for (stageId in ["1", "2"]) {
+      def bakeStage = pipeline.stageById(stageId)
+      def graph = StageGraphBuilder.beforeStages(bakeStage)
+      new BakeStage(regionCollector: new RegionCollector()).beforeStages(bakeStage, graph)
+      def childBakeStages = graph.build()
+      childBakeStages.eachWithIndex { it, idx ->
+        it.context.ami = "${idx}"
+        it.context.imageName = "image-from-bake-stage-${stageId}"
+      }
+      pipeline.stages.addAll(childBakeStages)
+    }
+
+    dynamicConfigService.isEnabled("stages.bake.failOnImageNameMismatch", false) >> { true }
+
+    when:
+    new BakeStage.CompleteParallelBakeTask(dynamicConfigService).execute(pipeline.stageById("1"))
+
```

Comment:
Minor... You expect some other exception to be thrown here ? If not we can say noExceptionThrown() here.

Reply:
I was advised to use a specific exception from kork for this case, so here I'm looking for that specific exception. If any other type of exception is thrown, it's unexpected in this test and it should fail.